### PR TITLE
Added Invoke-CdnContentPurge.ps1 plus Associated Unit Tests +semver: minor

### PR DESCRIPTION
**Pull request by mwevill**
- New Script Invoke-CdnContentPurge.ps1 to clear all content from a cdn.

- Associated Unit Tests

- Updated Unit.Tests.Config.json with entries to allow test.

### CODE REVIEW

File: Infrastructure-Scripts/Invoke-CdnContentPurge.ps1 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    [Parameter(Mandatory = $true)]
+    [String]$CDNProfileName,
+    [Parameter(Mandatory = $true)]
+    [String]$CDNEndPointName,
+    [Parameter(Mandatory = $false)]
+    [String]$PurgeContent = ""
+)
+try {
+    if ( $PurgeContent -eq "" ) {
+        throw "Purge Content blank will not run purge"
```
[the remark this answers is not in the document]
Comment (mwevill) [reply Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: Resolved

File: Infrastructure-Scripts/Invoke-CdnContentPurge.ps1 [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    The CDN EndPoint Name
+
+    .PARAMETER PurgeContent
+    The assest you wish to purge from the edge nodes
+    Single URL Purge: Purge individual asset by specifying the full URL, e.g., "/pictures/image1.png" or "/pictures/image1"
+    Wildcard purge: Purge all folders, sub-folders, and files under an endpoint with "/*"  e.g. "/* " or "/pictures/*"
+    Root domain purge: Purge the root of the endpoint with "/" in the path
+
+    .EXAMPLE
+
```
[the remark this answers is not in the document]
Comment (mwevill) [reply Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: Removed

File: Infrastructure-Scripts/Invoke-CdnContentPurge.ps1 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    [String]$PurgeContent = ""
+)
+try {
+    if ( $PurgeContent -eq "" ) {
+        throw "Purge Content blank will not run purge"
+    }
+    # --- Set CDN EndPoint
+    $CDNEndpoint = Get-AzCdnEndpoint -ResourceGroupName $CDNProfileResourceGroup -ProfileName $CDNProfileName -EndpointName $CDNEndpointName
+    if (!$CDNEndpoint) {
+        throw "CDN Endpoint Does not exist"
```
[the remark this answers is not in the document]
Comment (mwevill) [reply Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: Corrected

File: Infrastructure-Scripts/Invoke-CdnContentPurge.ps1 [hunk 4]
```diff
@@ -32,7 +31,7 @@ Param(
     [String]$CDNProfileName,
     [Parameter(Mandatory = $true)]
     [String]$CDNEndPointName,
-    [Parameter(Mandatory = $false)]
+    [Parameter(Mandatory = $true)]
     [String]$PurgeContent = ""
```
Comment (adam230594) [inline Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: No longer needs the default to ""

Comment (mwevill) [reply Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: if ( $PurgeContent -eq "" ) { } needed now if it is now a mandatory parameter? Still needed as we don't want to run a purge content with a blank pattern match. Waste of resources.

File: Infrastructure-Scripts/Invoke-CdnContentPurge.ps1 [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    [String]$CDNProfileResourceGroup,
+    [Parameter(Mandatory = $true)]
+    [String]$CDNProfileName,
+    [Parameter(Mandatory = $true)]
+    [String]$CDNEndPointName,
+    [Parameter(Mandatory = $true)]
+    [String]$PurgeContent
+)
+try {
+    if ( $PurgeContent -eq " " ) {
```
Comment (adam230594) [inline Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]:
```suggestion
    if ( $PurgeContent.Trim() -eq "" ) {
```

Comment (adam230594) [reply Infrastructure-Scripts/Invoke-CdnContentPurge.ps1]: I think this would be more dynamic

**Pull request merged**